File: reverse_polish/reverse_polish.js
String.prototype.reverse = function() {
    return this.split('').reverse().join('');
};

var ReversePolish = {
    op: ["*", ""],
    start: function() {
        for (var i = 1000; i < 10000; i++) {
            var c = String(i);
            for (var j = 0, len = this.op.length; j < len; j++) {
                for (var k = 0; k < len; k++) {
                    for (var l = 0; l < len; l++) {
                        var val = Number(c.charAt(3)) + this.op[j] + Number(c.charAt(2)) + this.op[k] + Number(c.charAt(1)) + this.op[l] + Number(c.charAt(0));
                        if (val.length > 4) {
                            if (i == eval(val)) {
                                console.log(c.reverse() + " → " + val + " = " + i);
                            }
                        }
                    }
                }
            }
        }
    }
};

ReversePolish.start();

File: reverse_polish/reverse_polish.rb
module ReversePolish
  def self.start
    operator = ["*", ""]
    (1000..9999).each do |n|
      str_num = n.to_s
      operator.each do |i|
        operator.each do |j|
          operator.each do |k|
            str = "#{str_num[0]}#{i}#{str_num[1]}#{j}#{str_num[2]}#{k}#{str_num[3]}"
            next if str =~ /\*0/
            result = eval(str)
            puts "#{str_num} → #{str} = #{result}" if result == str_num.reverse.to_i && str.length > 4
          end
        end
      end
    end
  end
end

ReversePolish.start
